src/main/java/org/dcache/SapphireDriver.java

```diff
@@ -51,9 +53,15 @@ public void flush(Iterable<FlushRequest> requests)
         _log.debug("Triggered flush()");
 
         for(FlushRequest flushRequest : requests) {
-            flushRequest.activate();
-            _log.debug("Added file to flushRequestQueue");
-            flushRequestQueue.add(flushRequest);
+            try {
+                flushRequest.activate().get();
```

Comment:
What is get() doing?

Reply:
The `flushRequest.activate()` returns a ListenableFuture. The `get` method will waits until the future is successfully of exceptionally complets.